scripts/install_xcpng.py

```diff
@@ -230,11 +243,15 @@ def main():
         host2 = pool2.master
         host2.inventory = host2._get_xensource_inventory()
         check_mac_address(host2, mac_address)
-        logging.info('New host is started and enabled')
-        if host2.xcp_version != version.parse(xcp_version):
+        logging.info(f'Target host is started and enabled in version:{host2.xcp_version}')
+        if args.action == 'restore' and host2.xcp_version == version.parse(xcp_version):
+            raise Exception(
+                f"The installed host ({vm_ip_address}) is not in a previous version. Got: {host2.xcp_version}.\n"
+            )
+        elif args.action != 'restore' and host2.xcp_version != version.parse(xcp_version):
```

Comment:
I'm not sure about this condition in the case `args.action` is `upgrade` but I might misunderstand.